src/ui/pages/SwapForm/fieldsets/ReceiveTokenField/MarketAssetSelect/MarketAssetSelect.tsx

```diff
@@ -42,16 +42,24 @@ export function MarketAssetSelect({
       ),
     [addressPositions]
   );
+
+  const { data: popularAssetCodes, isLoading: popularAssetCodesAreLoading } =
+    useQuery({
+      queryKey: ['getPopularTokens', chain],
+      queryFn: () => getPopularTokens(chain),
+      suspense: false,
+      retry: false,
+      staleTime: Infinity,
+    });
+
   const { networks } = useNetworks();
   const nativeAssetId = chain
     ? networks?.getNetworkByName(chain)?.native_asset?.id
     : ETH;
+
   const { data: popularAssetsResponse } = useAssetsPrices({
     currency,
-    asset_codes: [
-      nativeAssetId !== ETH ? nativeAssetId : null,
-      ...popularAssetsList,
-    ].filter(Boolean) as string[],
+    asset_codes: popularAssetCodes || [nativeAssetId || ETH],
```

Comment:
Not sure if these are equivalent, I mean they could possibly produce different lists of `asset_codes`:
1. `nativeAssetId !== ETH ? nativeAssetId : null` –  excludes ETH when `nativeAssetId` is ETH
2. `popularAssetCodes || [nativeAssetId || ETH]` – will include ETH if `nativeAssetId` is "falsy"

Reply:
I made ETH as a fallback value anyway, cause we shouldn't show empty token field. If we don't have popular tokens, native token and eth doesn't exist on chain, it is a corner case that I don't want to cover on our side :)